apis/vpcresources/v1beta1/vpcresourcerequest_types.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package v1beta1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// Important: Run "make" to regenerate code after modifying this file
+
+// VPCResourceRequestSpec defines the desired state of VPCResourceRequest
+type VPCResourceRequestSpec struct {
+	RequestTypes  map[string]ResourceType `json:"requestTypes,omitempty"`
+	RequestTarget Target                  `json:"requestTarget,omitempty"`
```

Comment:
Consider removing request, to avoid stuttering. e.g. VPCResourceRequest.RequestType vs VPCResourceRequest.Type
```suggestion
	Types  map[string]ResourceType `json:"types,omitempty"`
	Target Target                  `json:"target,omitempty"`
```

Reply:
Makes sense, Ellis. Will update the naming.